### DISCUSSION TITLE
Ability to query entities from SDK

### DISCUSSION THREAD

**Discussion by Pranov1** [langfuse/langfuse#1043] [Ideas]
Can that ability to query entities like traces, spans, and generations be added to the sdk? Something like get_traces(name=“RUN_XYZ_*“, metadata=...) bonus points for regex support.

Similar capabilities exist currently in the UI

**Comment by marcklingen** [comment]
Agree, the current public api and python wrappers have limitations and the public api is independent from the trpc API which powers the frontend.

A graphql interface to flexibly query/aggregate data from langfuse could be interesting

**Comment by marcklingen** [comment]
We have added this recently to cover more of the API with the SDKs: https://langfuse.com/docs/query-traces

No full coverage yet + no fully dynamic query builder. Keeping this thread open.
